torchsde/_core/sdeint.py

```diff
@@ -146,8 +145,9 @@ def check_contract(sde, method, logqp, ts, y0, bm, names, adjoint_method=None):
             raise ValueError(f"Drift must return a Tensor of the same shape as y0. Got drift shape {drift_shape_} but "
                              f"y0 shape {y0_.shape}.")
 
+    # TODO: Think more carefully about scalar noise shape checks.
     noise_channels = diffusion_shape[0][-1]
-    if sde.noise_type in (NOISE_TYPES.additive, NOISE_TYPES.general, NOISE_TYPES.scalar):
+    if sde.noise_type in (NOISE_TYPES.additive, NOISE_TYPES.general):
```

Comment:
I'm a bit uncertain about removing a check. What's currently checked is what matches the documentation. (But not the examples or tests I don't think: I think those are what we should update.)

Reply:
The current code doesn't work when diffusion outputs (..., 1). The temporary removal of scalar is to ensure that the tests pass. I will fix this in another PR, as it may require some other non-trivial changes in the registered functions and `step`.